src/CTA.WebForms2Blazor/ControlConverters/ControlConverter.cs

```diff
@@ -21,22 +28,80 @@ public virtual HtmlNode Convert2Blazor(HtmlNode node)
             return Convert2BlazorFromParts(NodeTemplate, BlazorName, ConvertAttributes(node.Attributes), node.InnerHtml);
         }
 
-        private string ConvertAttributes(HtmlAttributeCollection attributeCollection)
+        protected string ConvertAttributes(HtmlAttributeCollection attributeCollection)
         {
             var convertedAttributes = attributeCollection
                 .Where(attr => AttributeMap.ContainsKey(attr.Name))
-                .Select(attr => $"{AttributeMap[attr.Name]}={attr.Value}");
-            
+                .Select(attr => $"{AttributeMap[attr.Name]}='{attr.Value}'");
+
             var convertedAttributesString = string.Join(" ", convertedAttributes);
             
             return convertedAttributesString;
         }
 
-        private HtmlNode Convert2BlazorFromParts(string template, string name, string attributes, string body)
+        protected HtmlNode Convert2BlazorFromParts(string template, string name, string attributes, string body)
         {
             string newContent = String.Format(template, name, attributes, body);
             HtmlNode newNode = HtmlNode.CreateNode(newContent);
             return newNode;
         }
+
+        public static HtmlDocument ConvertEmbeddedCode(HtmlDocument document)
+        {
+            var documentNode = document.DocumentNode;
+            var htmlString = documentNode.WriteTo();
+            
+            var dataBindMatcher = DataBindRegex.Matches(htmlString);
+            var singleExprMatcher = SingleExpRegex.Matches(htmlString);
+            var directiveMatcher = DirectiveRegex.Matches(htmlString);
+            var aspExprMatcher = AspExpRegex.Matches(htmlString);
+            
+            //Might be a better way to do this
```

Comment:
Might want to consider using `Regex.Replace()` instead of `Regex.Match()` followed by `String.Replace()`. I think this would eliminate the need for these loops and each one might be doable in a single line

Reply:
Oh cool, I wasn't sure that was a thing but I'll give a try

Reply:
Agreed and worse case you could still one line these by stringing together your current operations. DataBindRegex.Matches(htmlString).When(x => x.Success == true).Replace(x.Value, "@(" + x.Groups["expr"].Value + ")"); Something like the above.